### PR TITLE
Add new image-namespace argument and use value when building

### PR DESCRIPTION
**Pull request by groeges**
Add new image-namespace argument and use value when building

Signed-off-by: Steve Groeger GROEGES@uk.ibm.com

### CODE REVIEW

File: cmd/stack_package.go [hunk 1]
```diff
@@ -230,6 +232,9 @@ func newStackPackageCmd(rootConfig *RootCommandConfig) *cobra.Command {
 				return errors.Errorf("Error reading directories: %v", err)
 			}
 
+			repoDirName := filepath.Base(filepath.Dir(stackPath))
```
Comment (kylegc) [inline cmd/stack_package.go]: We can't count on the repo name to be in the path. A user can run this command from any path in their filesystem.

File: cmd/stack_package.go [hunk 2]
```diff
@@ -242,7 +247,7 @@ func newStackPackageCmd(rootConfig *RootCommandConfig) *cobra.Command {
 				Debug.Log("sourceDir is: ", sourceDir)
 
 				// create name for the tar files
-				versionedArchive := filepath.Join(devLocal, stackID+".v"+stackYaml.Version+".templates.")
+				versionedArchive := filepath.Join(devLocal, repoDirName+"."+stackID+".v"+stackYaml.Version+".templates.")
```
Comment (kylegc) [inline cmd/stack_package.go]: I think we discussed that the CI scripts would prepend the repo name with the `add-to-repo --release-url` option.
Does that sound familiar?

File: cmd/stack_package.go [hunk 3]
```diff
@@ -102,7 +104,7 @@ func newStackPackageCmd(rootConfig *RootCommandConfig) *cobra.Command {
 			appsodyHome := getHome(rootConfig)
 			Debug.Log("appsodyHome is:", appsodyHome)
 
-			devLocal := filepath.Join(appsodyHome, "stacks", "dev.local")
+			devLocal := filepath.Join(appsodyHome, "stacks", imageNamespace)
```
Comment (kylegc) [inline cmd/stack_package.go]: I'm concerned about this. I know we had a discussion about whether or not the archives should always be stored in `~/.appsody/stacks/dev.local` or in separate dirs, but I don't think we came to a conclusion. 

With your code here, I believe we would be putting the `dev.local-index.yaml` in `~/.appsody/stacks/imageNamespace`. Therefore there could be multiple `dev.local-index.yaml`s around, and they won't get merged properly.

In my opinion, we should leave this hard coded to "dev.local" _for now._ This seems safest as we continue to develop the new `add-to-repo` command functionality.

**Comment by groeges** [comment]
@kylegc Have made the changes you suggested and pushed the latest code. 
Could you please review again and see that everything is OK.

**Pull request merged**